### PR TITLE
Support for r3 instance types.

### PR DESCRIPTION
**Pull request by pwendell**
Amazon's new r3 instances come without EXT3-formatted drives. This patch
adds a workaround that detects this situation and formats the drives
as ext3.

Hopefully this will be subsumed later if anyone wants to pick up
SPARK-1079.

### CODE REVIEW

**Comment by shivaram** [comment]
Since these are SSDs why don't we just format it as ext4 or XFS ? (http://superuser.com/questions/228657/which-linux-filesystem-works-best-with-ssd might be helpful)

**Comment by pwendell** [comment]
Good call, let's format these as XFS. I'd bet ext3 works pretty badly on these :)

**Comment by pwendell** [comment]
@shivaram I updated this

File: setup-slave.sh [hunk 1]
```diff
@@ -18,6 +18,23 @@ echo "Setting up slave on `hostname`..."
...
+
+  rm -rf /mnt*
+  mkdir /mnt
+  mkfs.xfs /dev/sdb
+  mount -o $XFS_MOUNT_OPTS /dev/sdb /mnt
+
+  if [[ $instance_type == "r3.8xlarge" ]]; then
+    mkdir /mnt2
+    mkfs.xfs /dev/sdc
+    mount -o $XFS_MOUNT_OPTS /dev/sdb /mnt2
```
Comment (shivaram) [inline setup-slave.sh]: The mount should be for /dev/sdc

**Comment by shivaram** [comment]
One comment, looks good other than that.

It would be good to investigate if there is a nice way to detect unformatted block devices using `lsblk` or `fdisk` etc. But this should suffice for now I guess

**Comment by pwendell** [comment]
Okay cool, thanks Shivaram. I'll merge this.

**Pull request merged**